File: backend/src/main/java/com/esliceu/proyectoglobal/entity/Usuario.java
package com.esliceu.proyectoglobal.entity;

import javax.persistence.*;
import java.util.List;

/**
 * Code created by: mmonteirocl
 * Email: miguelmonteiroclaveri@gmail.com
 * Date: 18/02/2020
 * Package: com.esliceu.proyectoglobal.entity
 * Project: BackEnd
 */
@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idusuario")
    private Long idusuario;

    @Column(name = "username", length = 45, nullable = false)
    private String username;

    @Column(name = "email", length = 100, nullable = false, unique = true)
    private String email;

    @Column(name = "password", length = 300, nullable = false)
    private String password;

    @Column(name = "nombre", length = 45, nullable = false)
    private String nombre;

    @Column(name = "apellidos", length = 80)
    private String apellidos;


    /*
     * Local - Google - Facebook ......
     * */
    @Column(name = "authMode", length = 20, nullable = false)
    private String authMode;

    @OneToMany(mappedBy = "usuario", cascade = CascadeType.ALL)
    private List<Post> posts;


    public Usuario() {
    }


    public Long getIdusuario() {
        return idusuario;
    }

    public void setIdusuario(Long idusuario) {
        this.idusuario = idusuario;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getAuthMode() {
        return authMode;
    }

    public void setAuthMode(String authMode) {
        this.authMode = authMode;
    }

    public List<Post> getPosts() {
        return posts;
    }

    public void setPosts(List<Post> posts) {
        this.posts = posts;
    }
}

File: backend-nodejs/src/model/Usuario.ts
import {Column, Model, Table} from 'sequelize-typescript';

@Table({
    tableName: "usuario",
    createdAt: false,
    updatedAt: false,
})
export class Usuario extends Model<Usuario> {

    @Column({
        field: 'idusuario',
        primaryKey: true,
        type: 'number'
    })
    idusuario: number | undefined;

    @Column({
        field: 'username',
        type: 'string'
    })
    username: string | undefined;

    @Column({
        field: 'email',
        type: 'string'
    })
    email: string | undefined;

    @Column({
        field: 'password',
        type: 'string'
    })
    password: string | undefined;

    @Column({
        field: 'nombre',
        type: 'string'
    })
    nombre: string | undefined;

    @Column({
        field: 'apellidos',
        type: 'string'
    })
    apellidos: string | undefined;

    @Column({
        field: 'auth_mode',
        type: 'string'
    })
    authMode: string | undefined;
}